File: benchmark/nodejs/count_set_bits.js
const start = Date.now();
let total = 0;
for (let n = 1; n <= 100000; n++) {
    let x = n;
    while (x > 0.5) {
        if (x % 2 >= 1) {
            total++;
        }
        x = x / 2;
    }
}
const end = Date.now();
console.log(`Total: ${total}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/src/ruby/count_set_bits.rb
n=100000;total=0;start=Time.now.to_f*1000
(1..n).each{|i|x=i;while x>0;total+=x%2;x/=2;end}
finish=Time.now.to_f*1000;puts"Result: #{total}";puts"Time: #{(finish-start).round}ms"